### DISCUSSION TITLE
XMOJ-bug: OI赛制正在进行的比赛中部分题目可以通过pid提交运行

### DISCUSSION THREAD

**Discussion by boomzero** [boomzero/quicksubmit#8] [General]
如果用cid会显示等待, 用pid xmoj就评测 @langningchen

<img width="990" alt="image" src="https://github.com/boomzero/quicksubmit/assets/85378277/395353e0-b294-4b6d-bba8-31826ddda051">

**Comment by boomzero** [comment]
当然是用quickSubmit

**Comment by boomzero** [comment]
这一bug严重影响了比赛的公平性, 应及时修复

**Comment by langningchen** [comment]
我都已经不用 XMOJ 了……

**Reply by boomzero** [reply]
...

**Comment by boomzero** [comment]
Which is a pity, because I cannot tell you how convenient qsm truly is, especially during competitions.

**Comment by langningchen** [comment]
And I think the OJTool is really convenient too! 😄

**Reply by boomzero** [reply]
@langningchen This convenient?

<img width="584" alt="image" src="https://github.com/boomzero/quicksubmit/assets/85378277/c675168d-b2b2-44cb-ba41-1463a843b19f">

finishes in <1s with no text to speech capabilities...

**Reply by langningchen** [reply]
?

**Reply by boomzero** [reply]
@langningchen ?

**Reply by boomzero** [reply]
No arguments or flags!

**Reply by langningchen** [reply]
Yes, because you make a different directory for each problem. And you get the PID from cwd.

**Comment by langningchen** [comment]
But it is not capable with VSCode Tasks, which can only passin arguments.

**Reply by boomzero** [reply]
emmm

**Reply by boomzero** [reply]
idk what you're talking about, I don't use VScode anyways (vscode in NOI Linux doesn't have the C++ extension installed anyways)

**Comment by boomzero** [comment]
Basically this is how CLion works

**Reply by boomzero** [reply]
<img width="823" alt="image" src="https://github.com/boomzero/quicksubmit/assets/85378277/3c962458-70b2-488d-a424-fe63ab2a0552">

**Comment by boomzero** [comment]
anyways, I still think passing the password for every invocation is not great @langningchen

**Comment by boomzero** [comment]
Also I can't figure out how to pass cid in OJtool

**Reply by boomzero** [reply]
which is not ideal

**Comment by langningchen** [comment]
So, can you help me fix the problem? Thanks.

**Reply by boomzero** [reply]
I would add cid

**Reply by boomzero** [reply]
but quicksubmit is written is javascript, not C++

**Reply by boomzero** [reply]
so I can't

**Reply by boomzero** [reply]
Without contest id, people can't submit to contests in xmoj

**Comment by langningchen** [comment]
Or could you add feature to submit on Luogu in quicksubmit?

**Reply by boomzero** [reply]
@langningchen That would violate luogu's EULA

<img width="980" alt="image" src="https://github.com/boomzero/quicksubmit/assets/85378277/1f48acb6-5b88-46d6-afef-ca17f1b88a22">
Did you request permission with OJTool?

https://help.luogu.com.cn/ula/luogu

**Reply by langningchen** [reply]
Nope, but it doesn't matter.

**Comment by boomzero** [comment]
Why not?...
@langningchen

**Comment by langningchen** [comment]
因为毕竟这个协议一般不出什么事情也每人管，而且我没有对锣鼓及相关服务的正常运行进行干扰、破坏、修改或施加其他不良影响

**Comment by boomzero** [comment]
Okay, maybe I will consider it. But not now. I am busy enough doing xmoj problems.

**Comment by boomzero** [comment]
edit: stupid Github rendering engine

**Comment by boomzero** [comment]
Maybe, but it could lead to the banning of your ~~precious luogu account~~

**Comment by langningchen** [comment]
I have two accounts. (

**Comment by langningchen** [comment]
![image](https://github.com/boomzero/quicksubmit/assets/81672884/75889d18-b76a-4692-afd6-d7494b6c9c39)

**Reply by boomzero** [reply]
I have 3... @langningchen 
<img width="1389" alt="Screenshot 2024-07-11 at 08 24 12" src="https://github.com/boomzero/quicksubmit/assets/85378277/7193d2f6-8384-40e2-a1a2-5384dde78958">

**Reply by langningchen** [reply]
所以不用担心封号

**Comment by boomzero** [comment]
封号不是大号小号一起封吗？Sent from my iPadOn Jul 11, 2024, at 14:11, Langning Chen ***@***.***> wrote:﻿
所以不用担心封号

**Comment by langningchen** [comment]
我觉得它联系不起来我的两个账号吧

**Comment by boomzero** [comment]
ip address

**Comment by langningchen** [comment]
Proxy

**Comment by boomzero** [comment]
Are you using tor? If not, there are still many ways to fingerprint and track you.Sent from my iPadOn Jul 12, 2024, at 10:16, Langning Chen ***@***.***> wrote:﻿
Proxy

**Comment by langningchen** [comment]
I believe I won't be tracked.

**Comment by boomzero** [comment]
You never know… The NSA is probably tracking you now...

**Comment by langningchen** [comment]
...

**Comment by boomzero** [comment]
This is true. The NSA records all network traffic. It just can’t decrypt them.

**Comment by langningchen** [comment]
量子计算机攻击？

**Comment by boomzero** [comment]
Yes, store and decrypt later
